Userland/Libraries/LibIMAP/Parser.cpp

```diff
@@ -270,7 +270,10 @@ ErrorOr<void> Parser::parse_untagged()
 ErrorOr<StringView> Parser::parse_quoted_string()
 {
     dbgln_if(IMAP_PARSER_DEBUG, "p: {}, parse_quoted_string()", m_position);
-    auto str = consume_while([](u8 x) { return x != '"'; });
+    auto start_position = m_position;
+    while (!at_end() && (m_buffer[m_position] != '"' || m_buffer[m_position - 1] == '\\'))
```

Comment:
Does IMAP support `\\` as an escape for `\`? If so, this gets `"\\"` wrong I think.

The usual way to write this kind of loop is

```
while (!at_end()) {
  switch (m_buffer[m_position]) {
  case '\\':
    ++m_position;
    if (at_end())
      return Error::from_string_literal("unterminated \\ escape");
    // For bonus points, switch m_buffer[m_position] to only allow valid escape chars. if IMAP has \x2F or \u001A, need to handle those here too
    ++m_position;
  case '"':
    break;
  default:
    ++m_position;
}
```

(This is also why it's usually a good idea to add tests first and then update them while changing code, to make sure no new regressions creep in, and to add interesting corner cases discovered while writing the PR 🙂)

Reply:
Updated, with spec comments